proxy/process.go

```diff
@@ -410,7 +414,20 @@ func (p *Process) ProxyRequest(w http.ResponseWriter, r *http.Request) {
 
 	// prevent new requests from being made while stopping or irrecoverable
 	currentState := p.CurrentState()
-	if currentState == StateFailed || currentState == StateShutdown || currentState == StateStopping {
+
```

Comment:
There is a chance here that the upstream os process is still running/resident in memory. Simply resetting the state to `StateStopped` to have `p.start()` is likely not enough. I think we'd need something that additionally makes sure it is no longer there with a SIGTERM, and with that doesn't work a SIGKILL. 

I have been considering a more robust attempted recovery behaviour for failed processes. That enhancement should be in a different PR not this one. 

Can you revert this behaviour change. We can also ensure that this PR stays more focused on the original intent.

Reply:
Yeah I hit that issue when I was testing my PRs changes (hence my attempt to fix it), I've reverted this now to leave things as they were.